### ISSUE TITLE
Rational-Numbers: >, <, ==, !=, >=, <= comparison tests?

### ISSUE THREAD

**Issue by rpottsoh** [exercism/problem-specifications#1481]
Would there be any interest in seeing test cases for comparing rational numbers?
```JSON
            {
              "description": "A half is less than three fourths",
              "property": "lessThan",
              "input": {
                "r1": [1, 2],
                "r2": [3, 4]
              },
              "expected": true
            },
            {
              "description": "Half is greater than a fourth",
              "property": "greaterThan",
              "input": {
                "r1": [1, 2],
                "r2": [1, 4]
              },
              "expected": true
            },
            {
              "description": "One is greater than or equal to a third",
              "property": "greaterThanOrEqual",
              "input": {
                "r1": [13, 13],
                "r2": [1, 3]
              },
              "expected": true
            },
            {
              "description": "Two of the same are equal",
              "property": "equal",
              "input": {
                "r1": [6, 1],
                "r2": [6, 1]
              },
              "expected": true
            },
            {
              "description": "Two different values are not equal",
              "property": "notEqual",
              "input": {
                "r1": [3, 4],
                "r2": [1, 2]
              },
              "expected": true
            }
```
so and so forth.

I looked briefly to see if this topic had already be discussed and didn't find anything.

**Comment by coriolinus** [comment]
suggestion re: input format:

```json
{
    "r1": {n: 3, d: 4},
    "r2": {n: 1, d: 2},
}
```

While it's true that technically certain rationals, imaginaries and the like, will have more than one term for the numerator and denominator, IMO it's better to assume that our rationals will have one of each, and name them instead of using an unbounded list. 

Widening focus: what's the point? Is this to teach people how to convert ints to floats?

**Comment by rpottsoh** [comment]
I like your suggestion to name the numerator and denominator.

For this specific exercise I am focusing on operator overloading.  If there is another or better exercise I should be considering I am not opposed to dropping the issue.  I can always implement more tests at my specific track instead of making changes to the canonical tests.

**Comment by coriolinus** [comment]
No, an operator overloading exercise sounds like a good idea; if your language supports it, it's often a convenient ability to have. It just wasn't obvious that that was the goal.

**Comment by SaschaMann** [comment]
On the Julia track we already test for the ordering, but not in the conventional way of exercism tests. We compare them to the results of the equivalent float operation instead of taking a handful of examples. This is both more thorough and easier to understand imo.

```julia
        @test (r == c) == (a / b == c)
        @test (r != c) == (a / b != c)
        @test (r <= c) == (a / b <= c)
        @test (r <  c) == (a / b <  c)
        @test (r >= c) == (a / b >= c)
        @test (r > c) == (a / b > c)
```
where r is a rational number and a, b, c are -5..5
https://github.com/exercism/julia/blob/master/exercises/rational-numbers/runtests.jl#L103-L129

**Comment by rpottsoh** [comment]
> No, an operator overloading exercise sounds like a good idea; if your language supports it, it's often a convenient ability to have. It just wasn't obvious that that was the goal.

I'm not looking to create a new exercise at this point.  Simply wondering about adding cases to the rational-numbers exercise. 

> Widening focus: what's the point? Is this to teach people how to convert ints to floats?

Not my intention.

**Comment by rpottsoh** [comment]
@SaschaMann 
> for a in -5:5, b in -5:5, c in -5:5

I am not familiar with `Julia` so please help me understand how this loop operates.  Is it like a single loop that has three iterators instead of a more typical single iterator or is this like having three nested loops?  I have deduced that the loop(s) are iterating from -5 to 5 and when a and b equal zero the loop is forced to iterate.  The same happens later in the `d` loop when c and d equal zero.

**Comment by SaschaMann** [comment]
@rpottsoh 

> I am not familiar with Julia so please help me understand how this loop operates. Is it like a single loop that has three iterators instead of a more typical single iterator or is this like having three nested loops?

Sorry, I should have made the example more clear for other languages. It's the latter, a shorter way to write three nested loops. E.g.:
```julia
julia> for a in 1:3, b in 1:3
           println(a, b)
       end
11
12
13
21
22
23
31
32
33
```

>  I have deduced that the loop(s) are iterating from -5 to 5 and when a and b equal zero the loop is forced to iterate. The same happens later in the d loop when c and d equal zero.

That's correct. 0/0 (as a rational number) throws an error, so it's skipped if it occurs anywhere, because it's not relevant to the ordering of the numbers. This special case might be handled differently in other languages (is 1/0 defined? If not -> skip it here). The 5 is chosen somewhat arbitrarily, 2 or 3 would likely be fine, too.

My main point: Testing these in a loop covers more cases at once without writing them all down explicitely and is, at least in my opinion, easier to read and understand.
